tests/integration/devfile/cmd_devfile_debug_test.go

```diff
@@ -67,20 +67,20 @@ var _ = Describe("odo devfile debug command tests", func() {
 			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), projectDirPath)
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-debugrun.yaml"), filepath.Join(projectDirPath, "devfile-with-debugrun.yaml"))
 			helper.RenameFile("devfile-with-debugrun.yaml", "devfile.yaml")
-			helper.CmdShouldPass("odo", "push", "--debug")
```

Comment:
since we are using context, we should also remove `helper.Chdir(context)` from beforeEach

Reply:
it doesn't matter because the context path here is the full path to the component but nevertheless i have updated it.

Reply:
> it doesn't matter because the context path here is the full path to the component but nevertheless if i remove helper.Chdir(context), I need to update a bunch of statements because we're copying devfiles from the repo to the context.

We have had problems before where the context flag wasn't working but it was never detected because we were changing into the context directory. The current directory might be used even though the context was passed.

>  We should be updating all the tests where we use context to not chdir to the dir if we want to do it!

Yes, we need this behavior to change and need a issue to track it.

Reply:
I have opened an issue here https://github.com/openshift/odo/issues/3801 and yes ideally we should not do it

Reply:
Thanks 👍